cpp/src/io/shp/polygon_shapefile_reader.cpp

```diff
@@ -26,16 +26,25 @@
 
 namespace {
 
+bool _reverse_winding = false;
+
 cudf::size_type read_ring(OGRLinearRing const& ring,
                           std::vector<double>& xs,
                           std::vector<double>& ys)
```

Comment:
```suggestion
                          std::vector<double>& ys,
                          const winding_order ring_order)
```

Reply:
Let's also replace raw loops with algorithms, and rely on OGR's iterator support. (Sadly they don't provide `rbegin()` and `rend()`, but presumably we can adapt using `std::reverse_iterator`.

The whole function would look something like this.

```
int read_ring(std::vector<point> const& ring,
                          std::vector<double>& xs,
                          std::vector<double>& ys,
                          winding_order ring_order)
{
  xs.reserve(ring.size());
  ys.reserve(ring.size());

  auto vert_to_tuple = [](auto vertex) { return thrust::make_tuple(vertex.getX(), vertex.getY()); };

  if (ring_order == winding_order::CLOCKWISE) {
    std::transform(ring.begin(), ring.end(), thrust::make_zip_iterator(thrust::make_tuple(xs.begin(), ys.begin())),
      vert_to_tuple);
  } else {
    std::transform(std::reverse_iterator(ring.end()), std::reverse_iterator(ring.begin()),
      thrust::make_zip_iterator(thrust::make_tuple(xs.begin(), ys.begin())), vert_to_tuple);
  }

  return ring.size();
}

Reply:
Note that below, where you call this function, you will need to reverse the direction at each level of nested rings (assuming we support multiple levels of nesting).  For this, I would suggest defining a convenience function:

```
winding_order reverse_winding(winding_order order) {
  return static_cast<winding_order>(!static_cast<bool>(order));
}
```